### PR TITLE
feat(github-actions): support vulnerability alerts

### LINKED ISSUE #17250: Support GitHub-native Dependabot Alerts for GitHub Actions packages
### What would you like Renovate to be able to do?

Renovate should support the GitHub-native Dependabot Alerts for GitHub Actions packages.

GitHub can now raise alerts when you're using a vulnerable GitHub Action runner. [^gh-blog] Relevant snippet from the `vulnerability Alerts` docs: [^renovate-docs]

> Renovate can read GitHub's Vulnerability Alerts to customize its Pull Requests.
> For this to work, you must enable the [Dependency graph](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#enabling-the-dependency-graph), and [Dependabot alerts](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-security-and-analysis-settings-for-your-repository).

Follow-up issue after discussion: https://github.com/renovatebot/renovate/discussions/17095

[^gh-blog]: [GitHub blog, Dependabot now alerts for vulnerable GitHub Actions](https://github.blog/2022-08-09-dependabot-now-alerts-for-vulnerable-github-actions/)
[^renovate-docs]: https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts

### If you have any ideas on how this should be implemented, please tell us here.

@viceice said we probably need to update these files:

https://github.com/renovatebot/renovate/blob/ab94b9503aaa3c28d523d9c7de6d06cf20b2d84f/lib/workers/repository/init/vulnerability.ts#L66-L73

https://github.com/renovatebot/renovate/blob/ab94b9503aaa3c28d523d9c7de6d06cf20b2d84f/lib/workers/repository/init/vulnerability.ts#L94-L102

### Is this a feature you are interested in implementing yourself?

No

### Reproduction repository

https://github.com/renovate-reproductions/17250-gh-actions-vulnerabillities

### PR DESCRIPTION
**Pull request by viceice**
## Changes
enable GitHub actions vulnerability alerts

## Context
- closes #17250
- https://github.com/renovate-reproductions/17250-gh-actions-vulnerabillities/issues/2

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
i can see the alert, but didn't get a pr
```
DEBUG: GitHub vulnerability details (repository=renovate-reproductions/17250-gh-actions-vulnerabillities)
       "alerts": {"actions/hashicorp/vault-action": {"< 2.2.0": "2.2.0"}}
DEBUG: alert package rules (repository=renovate-reproductions/17250-gh-actions-vulnerabillities)
       "alertPackageRules": [
         {
           "matchDatasources": ["github-tags"],
           "matchPackageNames": ["hashicorp/vault-action"],
           "matchCurrentVersion": "= 2.1.2",
           "matchFiles": [".github/workflows/build.yml"],
           "allowedVersions": "2.2.0",
           "prBodyNotes": [
             "### GitHub Vulnerability Alerts",
             "#### [CVE-2021-32074](https://nvd.nist.gov/vuln/detail/CVE-2021-32074)\n\nHashiCorp vault-action (aka Vault GitHub Action) before 2.2.0 allows attackers to obtain sensitive information from log files because a multi-line secret was not correctly registered with GitHub Actions for log masking.\n\nThe vault-action implementation did not correctly handle the marking of multi-line variables. As a result, multi-line secrets were not correctly masked in vault-action output.\n\nRemediation:\nCustomers using vault-action should evaluate the risk associated with this issue, and consider upgrading to vault-action 2.2.0 or newer. Please refer to https://github.com/marketplace/actions/vault-secrets for more information."
           ],
           "isVulnerabilityAlert": true,
           "force": {
             "groupName": null,
             "schedule": [],
             "dependencyDashboardApproval": false,
             "stabilityDays": 0,
             "rangeStrategy": "update-lockfile",
             "commitMessageSuffix": "[SECURITY]",
             "branchTopic": "{{{datasource}}}-{{{depName}}}-vulnerability",
             "prCreation": "immediate"
           }
         }
       ]
```

```
DEBUG: packageFiles with updates (repository=renovate-reproductions/17250-gh-actions-vulnerabillities, baseBranch=main)
       "config": {
         "github-actions": [
           {
             "packageFile": ".github/workflows/build.yml",
             "deps": [
               {
                 "depName": "hashicorp/vault-action",
                 "commitMessageTopic": "{{{depName}}} action",
                 "datasource": "github-tags",
                 "versioning": "docker",
                 "depType": "action",
                 "replaceString": "hashicorp/vault-action@v2.1.2",
                 "autoReplaceStringTemplate": "{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # tag={{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}",
                 "currentValue": "v2.1.2",
                 "depIndex": 0,
                 "updates": [
                   {
                     "bucket": "minor",
                     "newVersion": "v2.4.2",
                     "newValue": "v2.4.2",
                     "releaseTimestamp": "2022-08-15T18:56:52.000Z",
                     "newMajor": 2,
                     "newMinor": 4,
                     "updateType": "minor",
                     "branchName": "renovate-test/hashicorp-vault-action-2.x"
                   }
                 ],
                 "warnings": [],
                 "sourceUrl": "https://github.com/hashicorp/vault-action",
                 "currentVersion": "v2.1.2",
                 "isSingleVersion": true,
                 "fixedVersion": "v2.1.2"
               },
               {
                 "depName": "hashicorp/vault-action",
                 "commitMessageTopic": "{{{depName}}} action",
                 "datasource": "github-tags",
                 "versioning": "docker",
                 "depType": "action",
                 "replaceString": "hashicorp/vault-action@5e5c06a3c8e96b7c4757fe7a10e03469cdbd07bb # tag=v2.1.2",
                 "autoReplaceStringTemplate": "{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # tag={{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}",
                 "currentValue": "v2.1.2",
                 "currentDigest": "5e5c06a3c8e96b7c4757fe7a10e03469cdbd07bb",
                 "depIndex": 1,
                 "updates": [
                   {
                     "bucket": "minor",
                     "newVersion": "v2.4.2",
                     "newValue": "v2.4.2",
                     "releaseTimestamp": "2022-08-15T18:56:52.000Z",
                     "newMajor": 2,
                     "newMinor": 4,
                     "updateType": "minor",
                     "newDigest": "7d98524254c38dc6892804f991526ff30905f643",
                     "branchName": "renovate-test/hashicorp-vault-action-2.x"
                   }
                 ],
                 "warnings": [],
                 "sourceUrl": "https://github.com/hashicorp/vault-action",
                 "currentVersion": "v2.1.2",
                 "isSingleVersion": true,
                 "fixedVersion": "v2.1.2"
               }
             ]
           }
         ]
       }
```

**Review by zharinov — DISMISSED** [review]

**Comment by rarkins** [comment]
Maybe `= 2.1.2` does not match `v2.1.2` when default `semver` versioning is used?

**Comment by viceice** [comment]
> Maybe `= 2.1.2` does not match `v2.1.2` when default `semver` versioning is used?

yes, i think so too. maybe use npm versioning instead? 🤔

File: lib/workers/repository/init/vulnerability.ts [hunk 1]
```diff
@@ -43,6 +44,26 @@ type CombinedAlert = Record<
...
+};
+const datasourceMapping: Record<string, string> = {
+  COMPOSER: PackagistDatasource.id,
+  MAVEN: MavenDatasource.id,
+  NPM: NpmDatasource.id,
+  NUGET: NugetDatasource.id,
+  PIP: PypiDatasource.id,
+  RUBYGEMS: RubyGemsDatasource.id,
+  RUST: CrateDatasource.id,
+  ACTIONS: GithubTagsDatasource.id, // GitHub actions
```
Comment (HonkingGoose) [inline lib/workers/repository/init/vulnerability.ts:64]:
```suggestion
  ACTIONS: GithubTagsDatasource.id, // GitHub Actions
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Pull request closed without merging**